sdk/go/common/resource/asset.go

```diff
@@ -577,6 +582,10 @@ func (a *Archive) Serialize() map[string]interface{} {
 			}
 		}
 		result[ArchiveAssetsProperty] = assets
+	} else {
+		// Ensure that the assets property is always present, even if empty.
```

Comment:
The JSON/YAML attributes have omitempty so this would just get dropped anyway. But that's expected we only want the field when this _is_ an asset archive, and if assets is nil then it isn't. i.e. just remove this block.

Reply:
I didn't notice those since we are dealing with a custom serialization/deserialization here but indeed, this seems redundant so removed this block